backend/arm64_ast.mli

```diff
@@ -214,11 +285,45 @@ module Instruction_name : sig
     | ADDV
 end
 
-module Operand : sig
-  type t
-end
-
 module DSL : sig
+  val reg_op : Reg.t -> Operand.t
+
+  val imm : int -> Operand.t
+
+  val imm_float : float -> Operand.t
+
+  val imm_nativeint : nativeint -> Operand.t
+
+  (* ARM symbol operand; string must be converted to the OS specific
+     representation first *)
+  (* An ARM symbol can be used for both labels and symbols from the symbol
+     table; the respective conversion must be applied first *)
+  val symbol : Symbol.t -> Operand.t
+
+  val immediate_symbol : Symbol.t -> Operand.t
+
+  val shift : Operand.Shift.t -> Operand.t
+
+  (* Note: Memory accesses are only allowed on X-registers and the stack
+     pointer *)
+  val mem : base:Reg.t -> Operand.t
+
+  val mem_offset : base:Reg.t -> offset:int -> Operand.t
+
+  val mem_symbol : base:Reg.t -> symbol:Symbol.t -> Operand.t
+
+  val mem_pre : base:Reg.t -> offset:int -> Operand.t
+
+  val mem_post : base:Reg.t -> offset:int -> Operand.t
+
+  (* CR sspies: probably these should be part of the instruction name instead *)
+  val cond : Instruction_name.Cond.t -> Operand.t
```

Comment:
They are used as both an operand and part of a mnemonic, depending on the instruction.